databroker/broker.py

```diff
@@ -722,6 +724,73 @@ def process(self, headers, func, fields=None, fill=False):
 
     get_fields = staticmethod(get_fields)  # for convenience
 
+    def export(self, headers, db, new_root=None, copy_kwargs=None):
+        """
+        Export a list of headers.
+
+        .. warning::
+
+            This does not yet support copying external files.
+
+        Parameters:
+        -----------
+        headers : databroker.header
+            one or more headers that are going to be exported
+        db : databroker.Broker
+            an instance of databroker.Broker class, which has
+            filestore (fs) and metadatastore (mds) attributes
+            that will be the target to export info
+        new_root : str
+            optional. root directory of files that are going to
+            be exported
+        copy_kwargs : dict or None
+            passed through to the ``copy_files`` method on FileStore;
+            None by default
+
+        Returns
+        ------
+        file_pairs : list
+            list of (old_file_path, new_file_path) pairs generated by
+            ``copy_files`` moethdo on FileStore.
+        """
+        if copy_kwargs is None:
+            copy_kwargs = {}
+        try:
+            headers.items()
+        except AttributeError:
+            pass
+        else:
+            headers = [headers]
+        file_pairs = []
+        for header in headers:
+            # insert mds
+            db.mds.insert_run_start(**header['start'].to_name_dict_pair()[1])
+            events = self.get_events(header)
+            for descriptor in header['descriptors']:
+                db.mds.insert_descriptor(**descriptor.to_name_dict_pair()[1])
+                for event in events:
+                    event = event.to_name_dict_pair()[1]
+                    # 'filled' is obtained from the descriptor, not stored
+                    # in each event.
+                    event.pop('filled', None)
+                    db.mds.insert_event(**event)
+            db.mds.insert_run_stop(**header['stop'].to_name_dict_pair()[1])
+            # insert fs
+            res_uids = self.get_resource_uids(header)
+            for uid in res_uids:
+                fps = self.fs.copy_files(uid, new_root=new_root, **copy_kwargs)
+                file_pairs.extend(fps)
+                res = self.fs.resource_given_uid(uid)
+                db.fs.insert_resource(res['spec'],
+                                      res['resource_path'],
+                                      res['resource_kwargs'],
+                                      root=new_root)
+                datums = self.fs.datum_gen_given_resource(uid)
```

Comment:
https://github.com/NSLS-II/filestore/pull/122 will make this method,`datum_gen_given_resource`, return the full datum document (including the ID which we need) instead of just the datum kwargs.